src/polynomial.nr

```diff
@@ -0,0 +1,71 @@
+use std::field::bn254::assert_lt;
+
+/// Polynomial structure
+pub struct Polynomial {
+    // Coefficients of the polynomial (Biggest degree at the first index)
+    pub coefficients: Vec<Field>,
+}
+
+impl Polynomial {
+    /// Creates a new polynomial
+    pub fn new(coefficients: Vec<Field>) -> Self {
+        Polynomial { coefficients }
+    }
+
+    /// Evaluate a polynomial at a point in the field.
+    pub fn eval(self, x: Field) -> Field {
+        {
+            let mut acc = self.coefficients.get(0);
+            for i in 1..self.coefficients.len() {
+                acc = acc * x + self.coefficients.get(i);
+            }
+            acc
+        }
+    }
+
+    /// Adds `upper_bound` to the coefficients of the polynomial and constrains them to be in the range `[0, 2*upper_bound]`.
+    pub fn range_check_1bound(self, upper_bound: u64) {
+        for coeff in 0..self.coefficients.len() {
+            assert_lt(self.coefficients.get(coeff), 2 * upper_bound as Field + 1);
```

Comment:
is this mul per `2 * upper_bound as Field + 1` correct?

Reply:
Yes, since assert_lt is not checking equality we are adding 1 to the operation. This way we simulate it as assert_little_or_equal